File: solutions/2317-count-collisions-on-a-road/count-collisions-on-a-road.py
class Solution(object):
    def countCollisions(self, directions):
        n = len(directions)
        left, right = 0, n - 1
        stationCount = 0
        if n == 1:
            return 0
        while left < n and directions[left] == "L":
            left += 1
        while right > 0 and directions[right] == "R":
            right -= 1
        for i in range(left, right + 1):
            car = directions[i]
            if car == "S":
                stationCount += 1                

        return right + 1 - left - stationCount

File: solutions/2317-count-collisions-on-a-road/count-collisions-on-a-road.js
var countCollisions = function (directions) {
    let res = 0;
    let flag = -1;
    for (let i = 0; i < directions.length; i++) {
        const c = directions[i];
        if (c === "L") {
            if (flag >= 0) {
                res += flag + 1;
                flag = 0;
            }
        } else if (c === "S") {
            if (flag > 0) {
                res += flag;
            }
            flag = 0;
        } else {
            if (flag >= 0) {
                flag++;
            } else {
                flag = 1;
            }
        }
    }
    return res;
};